openfl/display/MovieClip.hx

```diff
@@ -26,704 +28,996 @@ import openfl.text.TextField;
-						
-						if (__swf.symbols.exists (frameObject.symbol)) {
-							
-							symbol = __swf.symbols.get (frameObject.symbol);
-							displayObject = symbol.__createObject (__swf);
-							
-						}
-						
-						if (displayObject != null) {
-							
-							timelineObject = new TimelineObject (frameObject.id, frameObject.depth, frameObject.clipDepth, displayObject);
-							
-							if (frameObject.clipDepth > 0) __maskCount++;
-							__objectDepths.push (timelineObject);
-							__objects.set (frameObject.id, timelineObject);
-							
-							depthChange = true;
-							
-						}
-						
-					} else {
-						
-						timelineObject = __objects.get (frameObject.id);
-						displayObject = timelineObject.displayObject;
-						
-						if (timelineObject.displayObject.parent == null) {
-							
-							if (frameObject.clipDepth > 0) __maskCount++;
-							__objectDepths.push (timelineObject);
-							
-							depthChange = true;
-							
+					else {
+
+						for (frameObjectEntry in __indexFrameObjectEntryById) {
+
+							displayObjectCache = frameObjectEntry.displayObjectCache;
+
+							if (null != displayObjectCache.displayObject) {
+
+								if (displayObjectCache.characterId == frameObject.symbol
+									&& displayObjectCache.depth == frameObject.depth)
+								{
+
+									duplicate = displayObjectCache;
+									break;
+
+								}
+							}
+
 						}
-						
+
 					}
-					
-					if (displayObject != null) {
-						
-						__placeObject (displayObject, frameObject);
-						
+
+					if (null != duplicate) {
+
+						trace("Multiple CREATE tags for the same characterId and depth are being merged.\n"
+							+ "Asset \""+ untyped this.__swf.library.rootPath +"\", "
+							+ "MovieClip \""+ this.__name +"\",\n"
+							+ "existing: { "
+								+ "frame: "+ displayObjectCache.createFrameOneIndexed
+								+ ", frameObjectId: "+ displayObjectCache.createFrameObjectId
+								+ ", characterId: "+ displayObjectCache.characterId
+								+ ", depth: "+ displayObjectCache.depth
+							+" },\n"
+							+ "new: { "
+								+ "frame: "+ (frameZeroIndexed + 1)
+								+ ", frameObjectId: "+ frameObject.id
+								+ ", characterId: "+ frameObject.symbol
+								+ ", depth: "+ frameObject.depth
+							+" }");
+
+						displayObjectCache = duplicate;
+						displayObject = displayObjectCache.displayObject;
+
 					}
-					
-				} else {
-					
-					if (__objects.exists (frameObject.id)) {
-						
-						timelineObject = __objects.get (frameObject.id);
-						
-						if (timelineObject != null) {
-							
-							if (timelineObject.displayObject.parent == this) {
-								
-								removeChild (timelineObject.displayObject);
-								
+					else {
+
+						// instantiate child's DisplayObject first and only time
+						symbol = __swf.symbols.get (frameObject.symbol); // lookup Symbol class
+
+						if (null == symbol) {
+
+							trace("Unable to CREATE DisplayObject instance; got NULL symbol\n"
```

Comment:
once we figure out why null symbols exist (you may already know, i haven't had time to figure out yet), we should remove this unless/except debugging.